src/foam/nanos/ndiff/NDiffDAO.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+    package: 'foam.nanos.ndiff',
+    name: 'NDiffDAO',
+    extends: 'foam.dao.ProxyDAO',
+    javaIncludes: [
+        'foam.nanos.ndiff.NDiff',
+        'foam.dao.DAO',
+        'foam.dao.CompositeDAO',
+        'foam.core.X'
+    ],
+    javaCode: `
+        public NDiffDAO(foam.core.X x, foam.dao.DAO delegate) {
+            setX(x);
+            setDelegate(delegate);
+
+            foam.nanos.logger.Logger logger = (foam.nanos.logger.Logger) getX().get("logger");
+            logger.info("NDiffDAO: object created");
```

Comment:
This can be simplified with 
Loggers logger = Loggers.logger(x, this);
this will set up PrefixLogger with the class name, then you don't need to add "NDiffDAO" to each log call. 
Also, use  commas ',' rather than : and concatenation. 
logger.info("constructor");

Reply:
now using the Loggers API to get a logger